File: Cpp/Program/Vehicle.cpp
#pragma once
// Menggunakan pragma once agar header file hanya di-include satu kali

#include <bits/stdc++.h>
#include <iostream>
#include <string>
using namespace std;

// Deklarasi kelas Vehicle untuk merepresentasikan kendaraan
class Vehicle {
private:
    string namapemilik;      // Nama pemilik kendaraan
    string platNomor;        // Nomor plat kendaraan
    string merk;             // Merk kendaraan
    int tahunProduksi;       // Tahun produksi kendaraan
    string warna;            // Warna kendaraan
    int lebar;               // Lebar kendaraan dalam milimeter
    int panjang;             // Panjang kendaraan dalam milimeter

public:
    // Constructor dengan parameter untuk inisialisasi atribut kendaraan
    Vehicle(string namapemilik, string platNomor, string merk, int tahunProduksi, string warna, int lebar, int panjang) {
        this->namapemilik = namapemilik;
        this->platNomor = platNomor;
        this->merk = merk;
        this->tahunProduksi = tahunProduksi;
        this->warna = warna;
        this->panjang = panjang;
        this->lebar = lebar;
    }

    // Getter functions untuk mendapatkan nilai atribut kendaraan
    string getPlatNomor() {
        return this->platNomor;
    }

    string getMerk() {
        return this->merk;
    }

    string getPemilik() {
        return this->namapemilik;
    }

    int getTahunProduksi() {
        return this->tahunProduksi;
    }

    string getWarna() {
        return this->warna;
    }

    int getLebar() {
        return this->lebar;
    }

    int getPanjang() {
        return this->panjang;
    }

    // Setter functions untuk mengubah nilai atribut kendaraan
    void setPlatNomor(string platNomor) {
        this->platNomor = platNomor;
    }

    void setMerk(string merk) {
        this->merk = merk;
    }

    void setTahunProduksi(int tahunProduksi) {
        this->tahunProduksi = tahunProduksi;
    }

    void setWarna(string warna) {
        this->warna = warna;
    }

    void setPemilik(string pemilik) {
        this->namapemilik = pemilik;
    }

    void setLebar(int lbr) {
        this->lebar = lbr;
    }

    void setPanjang(int pjg) {
        this->panjang = pjg;
    }

    // Destructor
    ~Vehicle() {
    }
};

File: Python/Programs/Vehicle.py
class Vehicle:
    def __init__(self, namapemilik, platNomor, merk, tahunProduksi, warna, lebar, panjang):
        """
        Constructor method to initialize the Vehicle object.

        Parameters:
        - namapemilik (str): Owner's name
        - platNomor (str): License plate number
        - merk (str): Brand of the vehicle
        - tahunProduksi (int): Production year
        - warna (str): Color of the vehicle
        - lebar (float): Width of the vehicle
        - panjang (float): Length of the vehicle
        """
        self.namapemilik = namapemilik
        self.platNomor = platNomor
        self.merk = merk
        self.tahunProduksi = tahunProduksi
        self.warna = warna
        self.lebar = lebar
        self.panjang = panjang

    def getPlatNomor(self):
        """Get the license plate number."""
        return self.platNomor

    def getMerk(self):
        """Get the brand of the vehicle."""
        return self.merk

    def getPemilik(self):
        """Get the owner's name."""
        return self.namapemilik

    def getTahunProduksi(self):
        """Get the production year of the vehicle."""
        return self.tahunProduksi

    def getWarna(self):
        """Get the color of the vehicle."""
        return self.warna

    def getLebar(self):
        """Get the width of the vehicle."""
        return self.lebar

    def getPanjang(self):
        """Get the length of the vehicle."""
        return self.panjang

    def setPlatNomor(self, platNomor):
        """Set the license plate number."""
        self.platNomor = platNomor

    def setMerk(self, merk):
        """Set the brand of the vehicle."""
        self.merk = merk

    def setTahunProduksi(self, tahunProduksi):
        """Set the production year of the vehicle."""
        self.tahunProduksi = tahunProduksi

    def setWarna(self, warna):
        """Set the color of the vehicle."""
        self.warna = warna

    def setPemilik(self, pemilik):
        """Set the owner's name."""
        self.namapemilik = pemilik

    def setLebar(self, lbr):
        """Set the width of the vehicle."""
        self.lebar = lbr

    def setPanjang(self, pjg):
        """Set the length of the vehicle."""
        self.panjang = pjg

    # Destructor tidak diperlukan dalam Python
